sqlglot/parser.py

```diff
@@ -3184,10 +3184,17 @@ def _parse_types(
             elif self._match_text_seq("WITHOUT", "TIME", "ZONE"):
                 maybe_func = False
         elif type_token == TokenType.INTERVAL:
-            unit = self._parse_var()
+            span: t.Optional[t.List[exp.Expression]] = None
```

Comment:
switch this to an else and put the type on the first one

Reply:
Done